src/app/pages/receive/components/receive-tokens.tsx

```diff
@@ -1,28 +1,64 @@
+import { useMemo } from 'react';
+
 import { HomePageSelectors } from '@tests/selectors/home.selectors';
 import { css } from 'leather-styles/css';
 import { Stack } from 'leather-styles/jsx';
 
+import { isDefined } from '@shared/utils';
+
 import { copyToClipboard } from '@app/common/utils/copy-to-clipboard';
 import { useToast } from '@app/features/toasts/use-toast';
+import { useAlexSwap } from '@app/pages/swap/hooks/use-alex-swap';
+import { SwapAsset } from '@app/pages/swap/hooks/use-swap-form';
+import { sortSwappableAssetsBySymbol } from '@app/pages/swap/swap.utils';
+import { Avatar, defaultFallbackDelay, getAvatarFallback } from '@app/ui/components/avatar/avatar';
+import { Brc20AvatarIcon } from '@app/ui/components/avatar/brc20-avatar-icon';
 import { BtcAvatarIcon } from '@app/ui/components/avatar/btc-avatar-icon';
+import { RunesAvatarIcon } from '@app/ui/components/avatar/runes-avatar-icon';
+import { Src20AvatarIcon } from '@app/ui/components/avatar/src20-avatar-icon';
 import { StxAvatarIcon } from '@app/ui/components/avatar/stx-avatar-icon';
 
 import { receiveTabStyle } from '../receive-dialog';
 import { ReceiveItem } from './receive-item';
 
 interface ReceiveTokensProps {
   btcAddressNativeSegwit: string;
+  btcAddressTaproot: string;
   stxAddress: string;
   onClickQrBtc(): void;
   onClickQrStx(): void;
 }
 export function ReceiveTokens({
   btcAddressNativeSegwit,
+  btcAddressTaproot,
   stxAddress,
   onClickQrBtc,
   onClickQrStx,
 }: ReceiveTokensProps) {
   const toast = useToast();
+  const { createSwapAssetFromAlexCurrency, supportedCurrencies } = useAlexSwap();
+
+  const swappableAssets: SwapAsset[] = useMemo(
+    () =>
+      sortSwappableAssetsBySymbol(
+        supportedCurrencies.map(createSwapAssetFromAlexCurrency).filter(isDefined)
+      ),
+    [createSwapAssetFromAlexCurrency, supportedCurrencies]
+  );
+
+  const receivableAssets = useMemo(
+    () =>
+      swappableAssets
+        .filter(asset => asset.name !== 'STX')
+        .map(asset => ({
+          address: asset.principal.split('.')[0],
```

Comment:
@fbwoolf can you please verify if this is correct, that `address` is the first section of `principal`? 

I'm pretty sure it is but want to be certain

Reply:
think we have helper functions for logic like this

Reply:
This is the asset principal ...it has nothing to do with the address you would want the user to copy. For Stacks fungible tokens the address is just the Stacks address. It would be the same for all of them.